return_interface_cones.py

```diff
@@ -0,0 +1,54 @@
+#!/usr/bin/env python
+# coding: utf-8
+
+# In[ ]:
+
+
+def parse_cones_file(pathname, dictionary, switch):
+    '''Parses a .cones file and returns a formatted dictionary of all of its cones'''
+    pathname = pathname.strip() # eliminate embedded null character
+    if switch == "acceptorCones": # ACCEPTOR
+        with open(pathname, 'r') as f:
+            for line in f:
+                if line.startswith("CONES:"): #why do we skip lines starting with 'cones'? 
+                    break
+            for line in f:
+                coors = line.strip().split() # removes any leading and trailing whitespaces and split the resulting string
+                key = line.strip().split('[')[1][:-14] 
+                apex = tuple(coors[0:3])
+                perp_vec = tuple(coors [3:6])
+                dictionary['ACCEP ' + key] = [apex, perp_vec]
+    else: # DONOR 
+        with open(pathname, 'r') as f:
+            for line in f:
+                if line.startswith("CONES:"):
+                    break
+            for line in f:
+                coors = line.strip().split()
+                apex = tuple(coors[0:3])
+                perp_vec = tuple(coors [3:6])
+                key = line.strip().split("180.0")[1][12:]
+                dictionary['DONOR ' + key] = [apex, perp_vec]
+    return dictionary
+
+def return_condensed_dicts(half1_dict, half2_dict, half1_interface_atoms, half2_interface_atoms):
+    # Create empty dictionaries to store the filtered key-value pairs
+    new_half1_dict = {}
+    new_half2_dict = {}
+    
+    # Extract the tuples from the interface atoms arrays
+    half1_tuples = set(map(tuple, half1_interface_atoms)) # where are we getting these data from?
```

Comment:
half1_interface_atoms are returned from extract_lists. All functions I asked you to include in this file work together to perform an important overall function. You need to understand how they work together, i.e, how the output of one function is used as the input for another